packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -593,3 +595,45 @@ export function getValueTypeValue(
       return undefined;
   }
 }
+
+/**
+ * Get all of the root user-defined namespaces in a TypeSpec definition if they're tied to a client
+ * @param context
+ * @returns
+ */
+export function getRootUserDefinedNamespaceName(
+  context: TCGCContext,
+): [string, readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+  const rootNamespaces = [...context.program.getGlobalNamespaceType().namespaces.values()];
```

Comment:
a little bit confusing with the logic. what will `getGlobalNamespaceType` return?

Reply:
global namespace returns the step above all of the namespaces. For example, if you just have an empty typespec with imports to `TypeSpec` libraries, it will return an unnamed namespace at the root. It will then have `.namespaces = [TypeSpecNamespace]`. If you have something more robust, i.e. what we will see in the usages of the namespace flag,

```tsp
...
using TypeSpec.Http;
using Azure.ClientGenerator.Core;

namespace Communication {
 ...
}
```
it will return an unnamed namespace with `.namespaces = [TypeSpecNamespace, AzureNamespace, CommunicationNamespace]`.

What i'm trying to find here is the root namespace defined by the user